### ISSUE TITLE
Unnecessary parentheses in is pattern expression not reported

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#2398]
Following #2394, the following code results in a diagnostics:

```csharp
public class Foo
{
    public void Bar()
    {
        if ((new object() is bool b) && b) // SA1119
        {
            return;
        }
    }
}
```

When the code fix is applied, the parentheses are successfully removed, but a space is incorrectly added where the opening parenthesis was:

```csharp
public class Foo
{
    public void Bar()
    {
        if ( new object() is bool b && b)
        {
            return;
        }
    }
}
```

:memo: The `SA1119CSharp7UnitTests.TestPatternMatchingAsync` test can be updated when this is fixed rather than adding a new test for this.

**Comment by sharwell** [comment]
Starting with #2424, the diagnostic is no longer reported in this location. If fixed, it should apply to both the original is expressions and the new is pattern expression.

**Comment by vweijsters** [comment]
It seems to me that not reporting a diagnostic here is the correct behavior. Although the parenthesis are technically not needed, the readability is harmed when the parenthesis are removed. I would propose to close this issue as won't fix.

**Comment by dlemstra** [comment]
Could we change `SA1119CSharp7UnitTests.TestPatternMatchingAsync` to test for `EmptyDiagnosticResults` instead of the `Skip` it has now? I could submit a PR for this.

**Comment by sharwell** [comment]
@dlemstra 👍
